From 683baa44d9c49fa6792ac98361d5c8b2e291aadd Mon Sep 17 00:00:00 2001
From: Vaciller <wyc_19533480830@outlook.com>
Date: Sun, 12 Jul 2026 01:12:06 +0800
Subject: [PATCH 82/89] feat: enable Bluetooth OPP file transfer activities and
 rewrite notification with icon-based accept/decline for Rockchip
 compatibility

Change-Id: I5c3fcc85b1a5c718191d5175c45c7c2f7a26a4a1
---
 android/app/AndroidManifest.xml               |   4 +-
 .../opp/BluetoothOppNotification.java         | 149 +++++++++---------
 2 files changed, 79 insertions(+), 74 deletions(-)

diff --git a/android/app/AndroidManifest.xml b/android/app/AndroidManifest.xml
index 9484744656..f7decabc4d 100644
--- a/android/app/AndroidManifest.xml
+++ b/android/app/AndroidManifest.xml
@@ -399,7 +399,7 @@
              android:process="@string/process"
              android:theme="@android:style/Theme.Material.Light.Dialog"
              android:label="@string/bt_share_picker_label"
-             android:enabled="false"
+             android:enabled="true"
              android:exported="true">
             <intent-filter>
                 <action android:name="android.intent.action.SEND"/>
@@ -441,7 +441,7 @@
              android:process="@string/process"
              android:excludeFromRecents="true"
              android:theme="@style/dialog"
-             android:enabled="false">
+             android:enabled="true">
         </activity>
         <activity android:name="com.android.bluetooth.opp.BluetoothOppBtErrorActivity"
              android:process="@string/process"
diff --git a/android/app/src/com/android/bluetooth/opp/BluetoothOppNotification.java b/android/app/src/com/android/bluetooth/opp/BluetoothOppNotification.java
index af138f2179..936ea1e0fc 100644
--- a/android/app/src/com/android/bluetooth/opp/BluetoothOppNotification.java
+++ b/android/app/src/com/android/bluetooth/opp/BluetoothOppNotification.java
@@ -45,6 +45,7 @@ import android.net.Uri;
 import android.os.Handler;
 import android.os.Message;
 import android.os.Process;
+import android.os.SystemProperties;
 import android.text.format.Formatter;
 import android.util.Log;
 
@@ -564,78 +565,82 @@ class BluetoothOppNotification {
             Intent baseIntent = new Intent().setDataAndNormalize(contentUri)
                     .setClassName(mContext,
                             BluetoothOppReceiver.class.getName());
-            Notification.Action actionDecline =
-                    new Notification.Action.Builder(Icon.createWithResource(mContext,
-                            R.drawable.ic_decline),
-                            mContext.getText(R.string.incoming_file_confirm_cancel),
-                            PendingIntent.getBroadcast(mContext, 0,
-                                    new Intent(baseIntent).setAction(Constants.ACTION_DECLINE),
-                                    PendingIntent.FLAG_IMMUTABLE)).build();
-            Notification.Action actionAccept = new Notification.Action.Builder(
-                    Icon.createWithResource(mContext,R.drawable.ic_accept),
-                    mContext.getText(R.string.incoming_file_confirm_ok),
-                    PendingIntent.getBroadcast(mContext, 0,
-                            new Intent(baseIntent).setAction(Constants.ACTION_ACCEPT),
-                            PendingIntent.FLAG_IMMUTABLE)).build();
-            Notification public_n =
-                    new Notification.Builder(mContext, OPP_NOTIFICATION_CHANNEL).setOnlyAlertOnce(
-                            true)
-                            .setOngoing(true)
-                            .setWhen(info.mTimeStamp)
-                            .addAction(actionDecline)
-                            .addAction(actionAccept)
-                            .setContentIntent(PendingIntent.getBroadcast(mContext, 0,
-                                    new Intent(baseIntent).setAction(
-                                            Constants.ACTION_INCOMING_FILE_CONFIRM),
-                                    PendingIntent.FLAG_IMMUTABLE))
-                            .setDeleteIntent(PendingIntent.getBroadcast(mContext, 0,
-                                    new Intent(baseIntent).setAction(Constants.ACTION_HIDE),
-                                    PendingIntent.FLAG_IMMUTABLE))
-                            .setColor(mContext.getResources()
-                                    .getColor(
-                                            android.R.color
-                                                    .system_notification_accent_color,
-                                            mContext.getTheme()))
-                            .setContentTitle(mContext.getText(
-                                    R.string.incoming_file_confirm_Notification_title))
-                            .setContentText(fileNameSafe)
-                            .setStyle(new Notification.BigTextStyle().bigText(mContext.getString(
-                                    R.string.incoming_file_confirm_Notification_content,
-                                    info.mDeviceName, fileNameSafe)))
-                            .setSubText(Formatter.formatFileSize(mContext, info.mTotalBytes))
-                            .setSmallIcon(R.drawable.bt_incomming_file_notification)
-                            .setLocalOnly(true)
-                            .build();
-            Notification n =
-                    new Notification.Builder(mContext, OPP_NOTIFICATION_CHANNEL).setOnlyAlertOnce(
-                            true)
-                            .setOngoing(true)
-                            .setWhen(info.mTimeStamp)
-                            .setContentIntent(PendingIntent.getBroadcast(mContext, 0,
-                                    new Intent(baseIntent).setAction(
-                                            Constants.ACTION_INCOMING_FILE_CONFIRM),
-                                    PendingIntent.FLAG_IMMUTABLE))
-                            .setDeleteIntent(PendingIntent.getBroadcast(mContext, 0,
-                                    new Intent(baseIntent).setAction(Constants.ACTION_HIDE),
-                                    PendingIntent.FLAG_IMMUTABLE))
-                            .setColor(mContext.getResources()
-                                    .getColor(
-                                            android.R.color
-                                                    .system_notification_accent_color,
-                                            mContext.getTheme()))
-                            .setContentTitle(mContext.getText(
-                                    R.string.incoming_file_confirm_Notification_title))
-                            .setContentText(fileNameSafe)
-                            .setStyle(new Notification.BigTextStyle().bigText(mContext.getString(
-                                    R.string.incoming_file_confirm_Notification_content,
-                                    info.mDeviceName, fileNameSafe)))
-                            .setSubText(Formatter.formatFileSize(mContext, info.mTotalBytes))
-                            .setSmallIcon(R.drawable.bt_incomming_file_notification)
-                            .setLocalOnly(true)
-                            .setVisibility(Notification.VISIBILITY_PRIVATE)
-                            .setPublicVersion(public_n)
-                            .build();
-            mNotificationMgr.notify(NOTIFICATION_ID_PROGRESS, n);
+            if (SystemProperties.get("ro.target.product","box").equals("box")) {
+                mContext.sendBroadcast(new Intent(baseIntent).setAction(Constants.ACTION_INCOMING_FILE_CONFIRM));
+            } else {
+                Notification.Action actionDecline =
+                        new Notification.Action.Builder(Icon.createWithResource(mContext,
+                                R.drawable.ic_decline),
+                                mContext.getText(R.string.incoming_file_confirm_cancel),
+                                PendingIntent.getBroadcast(mContext, 0,
+                                        new Intent(baseIntent).setAction(Constants.ACTION_DECLINE),
+                                        PendingIntent.FLAG_IMMUTABLE)).build();
+                Notification.Action actionAccept = new Notification.Action.Builder(
+                        Icon.createWithResource(mContext,R.drawable.ic_accept),
+                        mContext.getText(R.string.incoming_file_confirm_ok),
+                        PendingIntent.getBroadcast(mContext, 0,
+                                new Intent(baseIntent).setAction(Constants.ACTION_ACCEPT),
+                                PendingIntent.FLAG_IMMUTABLE)).build();
+                Notification public_n =
+                        new Notification.Builder(mContext, OPP_NOTIFICATION_CHANNEL).setOnlyAlertOnce(
+                                true)
+                                .setOngoing(true)
+                                .setWhen(info.mTimeStamp)
+                                .addAction(actionDecline)
+                                .addAction(actionAccept)
+                                .setContentIntent(PendingIntent.getBroadcast(mContext, 0,
+                                        new Intent(baseIntent).setAction(
+                                                Constants.ACTION_INCOMING_FILE_CONFIRM),
+                                        PendingIntent.FLAG_IMMUTABLE))
+                                .setDeleteIntent(PendingIntent.getBroadcast(mContext, 0,
+                                        new Intent(baseIntent).setAction(Constants.ACTION_HIDE),
+                                        PendingIntent.FLAG_IMMUTABLE))
+                                .setColor(mContext.getResources()
+                                        .getColor(
+                                                android.R.color
+                                                        .system_notification_accent_color,
+                                                mContext.getTheme()))
+                                .setContentTitle(mContext.getText(
+                                        R.string.incoming_file_confirm_Notification_title))
+                                .setContentText(fileNameSafe)
+                                .setStyle(new Notification.BigTextStyle().bigText(mContext.getString(
+                                        R.string.incoming_file_confirm_Notification_content,
+                                        info.mDeviceName, fileNameSafe)))
+                                .setSubText(Formatter.formatFileSize(mContext, info.mTotalBytes))
+                                .setSmallIcon(R.drawable.bt_incomming_file_notification)
+                                .setLocalOnly(true)
+                                .build();
+                Notification n =
+                        new Notification.Builder(mContext, OPP_NOTIFICATION_CHANNEL).setOnlyAlertOnce(
+                                true)
+                                .setOngoing(true)
+                                .setWhen(info.mTimeStamp)
+                                .setContentIntent(PendingIntent.getBroadcast(mContext, 0,
+                                        new Intent(baseIntent).setAction(
+                                                Constants.ACTION_INCOMING_FILE_CONFIRM),
+                                        PendingIntent.FLAG_IMMUTABLE))
+                                .setDeleteIntent(PendingIntent.getBroadcast(mContext, 0,
+                                        new Intent(baseIntent).setAction(Constants.ACTION_HIDE),
+                                        PendingIntent.FLAG_IMMUTABLE))
+                                .setColor(mContext.getResources()
+                                        .getColor(
+                                                android.R.color
+                                                        .system_notification_accent_color,
+                                                mContext.getTheme()))
+                                .setContentTitle(mContext.getText(
+                                        R.string.incoming_file_confirm_Notification_title))
+                                .setContentText(fileNameSafe)
+                                .setStyle(new Notification.BigTextStyle().bigText(mContext.getString(
+                                        R.string.incoming_file_confirm_Notification_content,
+                                        info.mDeviceName, fileNameSafe)))
+                                .setSubText(Formatter.formatFileSize(mContext, info.mTotalBytes))
+                                .setSmallIcon(R.drawable.bt_incomming_file_notification)
+                                .setLocalOnly(true)
+                                .setVisibility(Notification.VISIBILITY_PRIVATE)
+                                .setPublicVersion(public_n)
+                                .build();
+                mNotificationMgr.notify(NOTIFICATION_ID_PROGRESS, n);
+            }
         }
         cursor.close();
     }
-- 
2.53.0

